ROSNET/DataModel/Connection.cs

```diff
@@ -11,62 +9,75 @@ namespace ROSNET.DataModel
     /// </summary>
     public class Connection
     {
-        public int ConnID { get; }
+        //Header fields in Connection record:
+        public int Conn { get; } //unique ID for each connection
         public string Topic { get; }
+
+        //Data fields in Connection record:
         public string OriginalTopic { get; private set; }
         public string Type { get; private set; }
         public string Md5sum { get; private set; }
-        public string MessageDefinition { get; private set;  }
+        public List<FieldValue> MessageDefinition { get; private set;  } //defines how to read the message data of messages corresponding with this connection
         public string CallerID { get; private set; }
         public string Latching { get; private set; }
 
+        //List of messages corresponding to this connection
+        public List<Message> Messages { get; private set; }
+
+
+        /// <summary>
+        /// Create a connection with conn and topic from connection record header
+        /// </summary>
+        public Connection(FieldValue conn, FieldValue topic)
+        {
+            Conn = BitConverter.ToInt32(conn.Value);
+            Topic = System.Text.Encoding.Default.GetString(topic.Value);
+            Messages = new List<Message>();
+        }
+
         /// <summary>
-        /// Create a connection from conn and topic from connection record header, as well as a BinaryReader in position to read the connection header
+        /// Sets data in connection
         /// </summary>
-        public Connection(int connID, string topic, BinaryReader reader)
+        public void SetData(byte[] originalTopic, byte[] type, byte[] md5sum, List<FieldValue> messageDefinition, byte[] callerID, byte[] latching)
         {
-            ConnID = connID;
-            Topic = topic;
-            SetData(reader);
+            this.OriginalTopic = System.Text.Encoding.Default.GetString(originalTopic);
+            this.Type = System.Text.Encoding.Default.GetString(type);
+            this.Md5sum = System.Text.Encoding.Default.GetString(md5sum);
+            this.MessageDefinition = messageDefinition;
+            this.CallerID = System.Text.Encoding.Default.GetString(callerID);
+            this.Latching = System.Text.Encoding.Default.GetString(latching);
         }
 
         /// <summary>
-        /// Helper function to extract data from connection header
+        /// Adds a Message object to the connection's list of messages
         /// </summary>
-        private void SetData(BinaryReader reader)
+        public void AddMessage(Message message)
```

Comment:
En litt overflødig methode, du har en public property for message du bare kan kalle add på.